### PR TITLE
feat #10584 : Supported DataModel in Dashboard Page

### PR DESCRIPTION
**Pull request by Ashish8689**
### Describe your changes:

Fixes  part #10584

- I worked on adding DataModel in the Dashboard service.
- The DataModel detail page container the Model Tab 

image

image

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/DataModels/ModelTab/ModelTab.component.tsx [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+import { ModelTabProps } from './ModelTab.interface';
+
+const ModelTab = ({
+  data,
+  isReadOnly,
+  hasEditDescriptionPermission,
+  hasEditTagsPermission,
+  onUpdate,
+}: ModelTabProps) => {
+  console.log(hasEditTagsPermission);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/DataModels/ModelTab/ModelTab.component.tsx:40]: Remove
cc: @karanh37

File: openmetadata-ui/src/main/resources/ui/src/components/DataModels/ModelTab/ModelTab.component.tsx [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    if (isUndefined(editContainerColumnTags)) {
+      setEditContainerColumnTags(record);
+      // Fetch tags and terms only once
+      if (tagList.length === 0 || tagFetchFailed) {
+        fetchTags();
+      }
+    }
+  };
+
+  const renderColumnDescription: CellRendered<Column, 'description'> = (
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/DataModels/ModelTab/ModelTab.component.tsx:109]: wrap all the function with useCallback hook

File: openmetadata-ui/src/main/resources/ui/src/rest/dashboardAPI.ts [hunk 3]
```diff
@@ -129,3 +129,22 @@ export const restoreDashboard = async (id: string) => {
...
+
+export const getDataModels = async (
+  service: string,
+  fields: string,
+  paging?: PagingWithoutTotal
+) => {
+  const response = await APIClient.get<{
+    data: ServicePageData[];
+    paging: Paging;
+  }>(`/dashboard/datamodels`, {
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/rest/dashboardAPI.ts:141]: You can use PaginResponse<ServicePageData[]>

File: openmetadata-ui/src/main/resources/ui/src/pages/DataModelPage/DataModelPage.component.tsx [hunk 4]
```diff
@@ -0,0 +1,470 @@
...
+import { serviceTypeLogo } from 'utils/ServiceUtils';
+import { getTagsWithoutTier, getTierTags } from 'utils/TableUtils';
+import { showErrorToast } from 'utils/ToastUtils';
+
+enum DATA_MODELS_DETAILS_TABS {
+  MODEL = 'model',
+  ACTIVITY = 'activityFeed',
+  SQL = 'sql',
+  LINEAGE = 'lineage',
+}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/DataModelPage/DataModelPage.component.tsx:67]: interface file

File: openmetadata-ui/src/main/resources/ui/src/pages/DataModelPage/DataModelPage.component.tsx [hunk 5]
```diff
@@ -0,0 +1,470 @@
...
+    }
+  };
+
+  const {
+    hasViewPermission,
+    hasEditDescriptionPermission,
+    hasEditOwnerPermission,
+    hasEditTagsPermission,
+    hasEditTierPermission,
+  } = useMemo(() => {
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/DataModelPage/DataModelPage.component.tsx:108]: nit: useMemo should be before handlers for good readability

**Comment by Ashish8689** [comment]
@chirag-madlani  Updated as per comments!

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**